packages/framework/client-logger/external-telemetry/src/factory/index.ts

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IContainer } from "@fluidframework/container-definitions";
+import { ContainerTelemetryManager } from "../container";
+import { ContainerEventTelemetryProducer } from "../container/telemetryProducer";
+import type { ApplicationInsights } from "@microsoft/applicationinsights-web";
+import { AppInsightsTelemetryConsumer } from "../common";
+
+/**
+ * Namespace for grouping things related to {@link (TelemetryManagerConfig:interface)}
+ *
+ * @beta
+ */
+export namespace TelemetryManagerConfig {
+	/**
+	 * Configuration details for the external telemetry consumer to be used by a Telemetry Manager
+	 * @beta
+	 */
+	export interface IConsumerConfig {
+		/**
+		 * The type of telemetry consumer. This helps typecheck what the concrete type of a given instance of IConsumerConfig is.
+		 */
+		type: ConsumerConfigType;
+	}
+
+	/**
+	 * Contains constants for the available types for {@link (TelemetryManagerConfig:namespace).IConsumerConfig."type"}
+	 * @beta
+	 */
+	export const ConsumerConfigTypes = {
+		/**
+		 * Identifier to be used for {@link (TelemetryManagerConfig:namespace).IConsumerConfig} of type {@link (TelemetryManagerConfig:namespace).AppInsightsConsumerConfig}
+		 */
+		APP_INSIGHTS: "APP_INSIGHTS",
+	} as const;
+
+	/**
+	 * Aggregate Type for all available types for {@link (TelemetryManagerConfig:namespace).IConsumerConfig."type"}
+	 * @beta
+	 */
+	export type ConsumerConfigType = (typeof ConsumerConfigTypes)[keyof typeof ConsumerConfigTypes];
+
+	/**
+	 * Configuration for using Azure App Insights as the telemetry consumer for a given telemetry manager
+	 * @beta
+	 */
+	export interface AppInsightsConsumerConfig extends IConsumerConfig {
+		type: "APP_INSIGHTS";
+		/** An instance of an Azure Application Insights client {@link @microsoft/applicationinsights-web#ApplicationInsights}
+		 * The instance must be initialed before being used in the config. That command should be  {@link @microsoft/applicationinsights-web#ApplicationInsights.loadAppInsights}
+		 * `applicationInsightsClient.loadAppInsights(); `
+		 */
+		appInsightsClient: ApplicationInsights;
+	}
+}
+
+/**
+ * Config object for creating one or more telemetry managers covering differents scopes of the Fluid Framework
+ *
+ * @beta
+ */
+export interface TelemetryManagerConfig {
+	/**
+	 * Configuration for creating an external telemetry manager for container related telemetry.
+	 */
+	containerTelemetry?: {
+		container: IContainer;
+		consumerConfig: TelemetryManagerConfig.IConsumerConfig;
+	};
+}
+
```

Comment:
Two things here... `consumerConfig` feels like it should go out of `containerTelemetry`, if that section is intended for things which are specifically about telemetry from a container vs telemetry from something else. The consumer is common to all of them.

**But** the second thing is that I think we should not ask for a configuration for the telemetry consumer. I think users of the package (including us) should create and completely configure the consumer they want to use and just pass it to the factory as an `ITelemetryConsumer` so the manager (or things further inside this package) can blindly call `consume()` on it. We probably don't want logic in the factory to know how to configure different consumers (e.g. AppInsights, something in Amazon, something in Google, something local, etc.). Like that we can also get rid of the namespace above entirely.

Reply:
I agree there is something that feels wrong here. I've reworked the config file. LMK what you think of the updated version